src/rust/inetstack/mod.rs

```diff
@@ -406,7 +412,10 @@ impl<const N: usize> InetStack<N> {
                     let result: Result<(), Fail> = future.await;
                     match result {
                         Ok(()) => {
-                            qtable_ptr.borrow_mut().free(&qd);
+                            qtable_ptr
+                                .borrow_mut()
+                                .free::<TcpQueue<N>>(&qd)
+                                .expect("queue should exist");
```

Comment:
To follow our guidelines, could you add a comment explaining why it is safe to call `expect()` here?